node_modules/gh-users/lib/api.js

```diff
@@ -324,6 +335,242 @@ var updateAdminStatus = module.exports.updateAdminStatus = function(ctx, userId,
+};
+
+/**
+ * Remove an uploaded user CSV file
+ *
+ * @param  {Object}         userCSV                 File object representing the uploaded CSV file as returned by express
+ * @param  {Function}       callback                Standard callback function
+ * @api private
+ */
+var _cleanUpCSVFile = function(userCSV, callback) {
+    if (userCSV && userCSV.file) {
+        fs.exists(userCSV.file, function(exists) {
+            if (exists) {
+                fs.unlink(userCSV.file, function(err) {
+                    if (err) {
+                        log().warn({'err': err, 'file': userCSV}, 'Could not remove the user import CSV file');
+                    }
+                    return callback();
+                });
+            } else {
+                return callback();
+            }
+        });
+    } else {
+        return callback();
+    }
+};
+
+/**
+ * Import a set of users
+ *
+ * @param  {Number}     appId                       The id of the application to import the users into
+ * @param  {String}     authenticationStrategy      The authentication strategy for the user. One of `local` or `shibboleth`
+ * @param  {Boolean}    [forceProfileUpdate]        Whether the user information should be updated, even when other user information is already present
+ * @param  {String[][]} userRows                    The users records to import
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @api private
+ */
+var _importUsers = function(appId, authenticationStrategy, forceProfileUpdate, userRows, callback) {
+    if (_.isEmpty(userRows)) {
+        return callback();
+
+    // Add a progress log statement every 50 imported users
+    } else if (userRows.length % 50 === 0) {
+        log().info({
+            'authenticationStrategy': authenticationStrategy,
+            'appId': appId
+        }, 'Importing users from CSV. ' + userRows.length + ' users left to import');
+    }
+
+    // Import the next user
+    var userRow = userRows.pop();
+    _importUser(appId, authenticationStrategy, forceProfileUpdate, userRow, function(err) {
+        // Swallow the error so that 1 bad user account doesn't prevent the others
+        // from being created. The error will have been logged already
+
+        // Recursively import the next user (if any)
+        _importUsers(appId, authenticationStrategy, forceProfileUpdate, userRows, callback);
+    });
+};
+
+/**
+ * Import a user
+ *
+ * @param  {Number}     appId                       The id of the application to import the user into
+ * @param  {String}     authenticationStrategy      The authentication strategy for the user. One of `local` or `shibboleth`
+ * @param  {Boolean}    [forceProfileUpdate]        Whether the user information should be updated, even when other user information is already present
+ * @param  {String[]}   userRow                     The user row to import
```

Comment:
Is it feasible to indicate what value should be at what index of the row in the documentation?